subworkflows/local/input_check.nf

```diff
@@ -20,7 +20,17 @@ workflow INPUT_CHECK {
     reads_in = Channel.fromSamplesheet(
         "input", // apparentely input maps to params.input...
         parameters_schema: 'nextflow_schema.json',
-        skip_duplicate_check: true).map {
+        skip_duplicate_check: true)
+    // Check that samplesheet does not contain more samples than sample limit
+    reads_in.collect()
+    .map { items ->
+            if ((items.size() > params.max_samples) && !(params.max_samples == 0)) { // Default max_samples is 0, which is equivalent to "no-limit"
+                error "Pipeline is being run with ${items.size()} items, which exceeds the limit of ${params.max_samples}"
+            }
+            return items
+        }
+    .flatten()
```

Comment:
Do you need `flatten()` here? Since you're not using the results of the `map` anywhere (line below is directly accessing the `reads_in` variable again.